### ISSUE TITLE
Rework $ProtocolEventBody

### ISSUE THREAD

**Issue by rmarx** [quicwg/qlog#192]
The new CDDL approach uses a CDDL extension point to be more strict about which values are allowed in the `Event:data` field. 

Currently, it's defined as:

```
; The ProtocolEventBody is any key-value map (e.g., JSON object)
; only the optional trigger field is defined in this document
$ProtocolEventBody /= {
    ? trigger: text
    * text => any
}
; event documents are intended to extend this socket by using:
; NewProtocolSpecificEvents = EventType1 / EventType2 / ... / EventTypeN
; $ProtocolEventBody /= NewProtocolSpecificEvents
```

However, this effectively still allows just about any JSON object as ProtocolEventBody, as we're just extending the $ProtocolEventBody.

If we want to be (slightly) more strict, we could do something like this:
```
; main doc
ProtocolEventBodyPrototype = {
    ? trigger: text
    * text => any
}

ProtocolEventBody = ProtocolEventBodyPrototype / $ProtocolEventBodies

; event docs
NewProtocolSpecificEvents = EventType1 / EventType2 / ... / EventTypeN
$ProtocolEventBodies /= NewProtocolSpecificEvents
```

Alternatively, we can just NOT allow the general form (the `ProtocolEventBodyPrototype`) and just include that as an example and require event bodies to provide at least 1 defined event.

**Comment by LPardue** [comment]
I don't see any problem with the current level of extensibility. What benefit(s) would being more restrictive bring?

**Comment by LPardue** [comment]
After understanding how to apply CDDL types better, I now support more rigidity in extensibility.

**Comment by rmarx** [comment]
Solved by #417
